src/lib.rs

```diff
@@ -132,21 +219,18 @@ impl MediaType {
     }
 
     /// Sets the charset parameter to the given charset and returns the old value if present.
-    pub fn set_charset(&mut self, charset: Charset) -> Option<String> {
-        self.parameters.insert("charset".to_owned(), charset.to_string())
+    pub fn set_charset(&mut self, charset: Charset) -> Option<Cow<'static, str>> {
+        self.parameters.insert("charset".into(), Cow::Owned(charset.to_string()))
```

Comment:
nit: I think `into` suffices in place of `Cow::Owned`, but I guess it's not bad to be explicit.